LICENSE

```diff
@@ -187,7 +187,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2017 Dell, Inc.
+   Copyright [yyyy] [name of copyright owner]
```

Comment:
shouldn't this be 2021 Canonical?

Reply:
I think it depends on who you ask :)

The official howto says the following:
> Copy the full Apache 2.0 license text into a LICENSE file.
Create a 'NOTICE' file specific to your product's details, and complying with the instructions below...
https://infra.apache.org/licensing-howto.html#step-by-step

So the LICENSE should be copied and NOTICE should be adapted. 

The placeholder in question is in the appendix of the LICENSE file. If we do replace it, the appendix text will then make no sense. We would then have to modify other parts of the appendix, resulting in more than just copying the LICENSE. I've seen projects going both ways (e.g. see [edgex-go](https://github.com/edgexfoundry/edgex-go/blob/main/LICENSE#L179) which has removed the appendix header but didn't replace the placeholders).

```
APPENDIX: How to apply the Apache License to your work.

      To apply the Apache License to your work, attach the following
      boilerplate notice, with the fields enclosed by brackets "[]"
      replaced with your own identifying information. (Don't include
      the brackets!)  The text should be enclosed in the appropriate
      comment syntax for the file format. We also recommend that a
      file or class name and description of purpose be included on the
      same "printed page" as the copyright notice for easier
      identification within third-party archives.

   Copyright [yyyy] [name of copyright owner]

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
```

The placeholder must be replaced when applying the notice to a file. E.g.: https://github.com/canonical/edgex-snap-hooks/blob/master/utils.go#L4-L15

Reply:
I created another PR to add the NOTICE file: https://github.com/canonical/edgex-snap-hooks/pull/22

Reply:
ah sorry I did not notice that this is itself the boiler plate :facepalm: this looks good in that case